contrib/internal/httptrace/inferred_proxy.go

```diff
@@ -0,0 +1,152 @@
+	// of the request in milliseconds. This value corresponds to the
+	// 'context.requestTimeEpoch' in AWS API Gateway, providing a timestamp
+	// for when the request was initiated.
+	ProxyHeaderStartTimeMs = "X-Dd-Proxy-Request-Time-Ms"
+
+	// ProxyHeaderPath is the header used to indicate the path of the
+	// request. This value corresponds to 'context.path' in AWS API Gateway,
+	// and helps identify the resource that the request is targeting.
+	ProxyHeaderPath = "X-Dd-Proxy-Path"
+
+	// ProxyHeaderHttpMethod is the header used to indicate the HTTP method
+	// of the request (e.g., GET, POST, PUT, DELETE). This value corresponds
+	// to 'context.httpMethod' in AWS API Gateway, and provides the method
+	// used to make the request.
+	ProxyHeaderHttpMethod = "X-Dd-Proxy-Httpmethod"
+
+	// ProxyHeaderDomain is the header used to indicate the AWS domain name
+	// handling the request. This value corresponds to 'context.domainName'
+	// in AWS API Gateway, which represents the custom domain associated
+	// with the API Gateway.
+	ProxyHeaderDomain = "X-Dd-Proxy-Domain-Name"
+
+	// ProxyHeaderStage is the header used to indicate the AWS stage name
+	// for the API request. This value corresponds to 'context.stage' in
+	// AWS API Gateway, and provides the stage (e.g., dev, prod, etc.)
+	// in which the request is being processed.
+	ProxyHeaderStage = "X-Dd-Proxy-Stage"
+)
+
+type proxyDetails struct {
+	spanName  string
+	component string
+}
+
+type proxyContext struct {
+	startTime       time.Time
+	method          string
+	path            string
+	stage           string
+	domainName      string
+	proxySystemName string
+}
+
+var (
+	supportedProxies = map[string]proxyDetails{
+		"aws-apigateway": {
+			spanName:  "aws.apigateway",
+			component: "aws-apigateway",
+		},
+	}
+)
+
+func extractInferredProxyContext(headers http.Header) (*proxyContext, error) {
+	_, exists := headers[ProxyHeaderStartTimeMs]
+	if !exists {
+		return nil, errors.New("proxy header start time does not exist")
+	}
+
+	proxyHeaderSystem, exists := headers[ProxyHeaderSystem]
+	if !exists {
+		return nil, errors.New("proxy header system does not exist")
+	}
+
+	if _, ok := supportedProxies[proxyHeaderSystem[0]]; !ok {
+		return nil, errors.New("unsupported Proxy header system")
+	}
+
+	pc := proxyContext{
+		method:          headers.Get(ProxyHeaderHttpMethod),
+		path:            headers.Get(ProxyHeaderPath),
+		stage:           headers.Get(ProxyHeaderStage),
+		domainName:      headers.Get(ProxyHeaderDomain),
+		proxySystemName: headers.Get(ProxyHeaderSystem),
+	}
+
+	startTimeUnixMilli, err := strconv.ParseInt(headers[ProxyHeaderStartTimeMs][0], 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing time string: %v", err)
+	}
+	pc.startTime = time.UnixMilli(startTimeUnixMilli)
+
+	return &pc, nil
+}
+
+func startInferredProxySpan(requestProxyContext *proxyContext, parent ddtrace.SpanContext, opts ...ddtrace.StartSpanOption) (tracer.Span, error) {
```

Comment:
```suggestion
func startInferredProxySpan(requestProxyContext *proxyContext, parent ddtrace.SpanContext, opts ...ddtrace.StartSpanOption) tracer.Span {
```

this function never returns an error